DuckDuckGo/AppDelegate.swift

```diff
@@ -360,24 +368,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return tabsModel
     }
 
-    private func makeHistoryManager() -> HistoryManager {
-        let historyManager = HistoryManager(privacyConfigManager: ContentBlocking.shared.privacyConfigurationManager,
-                              variantManager: DefaultVariantManager(),
-                              database: HistoryDatabase.make()) { error in
-            Pixel.fire(pixel: .historyStoreLoadFailed, error: error)
-            if error.isDiskFull {
-                self.presentInsufficientDiskSpaceAlert()
-            } else {
-                self.presentPreemptiveCrashAlert()
+    private func makeHistoryManager(_ appSettings: AppSettings,
+                                    _ internalUserDecider: InternalUserDecider,
+                                    _ privacyConfigManager: PrivacyConfigurationManaging = ContentBlocking.shared.privacyConfigurationManager)
+            -> HistoryManager {
+
+        let historyManager = HistoryManager(privacyConfigManager: privacyConfigManager,
+                                            variantManager: DefaultVariantManager(),
+                                            database: HistoryDatabase.make(),
+                                            internalUserDecider: internalUserDecider,
+                                            isEnabledByUser: appSettings.recentlyVisitedSites)
+
+        // Ensure we don't do this if the history is disabled in privacy confg
+        if historyManager.isHistoryFeatureEnabled() {
+            do {
+                try historyManager.loadStore(onCleanFinished: {
+                    // Do future migrations after clean has finished.  See macOS for an example.
+                })
+            } catch {
+                Pixel.fire(pixel: .historyStoreLoadFailed, error: error)
+                if error.isDiskFull {
+                    self.presentInsufficientDiskSpaceAlert()
+                } else {
+                    self.presentPreemptiveCrashAlert()
```

Comment:
Silly question: for features like this where the contents are intended to be transitory, should we take some stronger actions to try to recover, like wiping out the DB file and loading it again? If I were a user whose history store failed to load, I think I'd prefer that the history feature stopped working but the app as a whole functioned as usual.

(FWIW I think the current approach is fine, but just checking)

Reply:
I think that's a question for the broader team tbh.